src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventNotificationEmailManager.java

```diff
@@ -0,0 +1,170 @@
+import uk.ac.cam.cl.dtg.isaac.dto.content.ContentDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.eventbookings.DetailedEventBookingDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.users.RegisteredUserDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.users.UserSummaryDTO;
+import uk.ac.cam.cl.dtg.segue.api.Constants;
+import uk.ac.cam.cl.dtg.segue.api.managers.UserAccountManager;
+import uk.ac.cam.cl.dtg.segue.api.services.ContentService;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.NoUserException;
+import uk.ac.cam.cl.dtg.segue.comm.EmailManager;
+import uk.ac.cam.cl.dtg.segue.comm.EmailType;
+import uk.ac.cam.cl.dtg.segue.dao.SegueDatabaseException;
+import uk.ac.cam.cl.dtg.segue.dao.content.ContentManagerException;
+import uk.ac.cam.cl.dtg.segue.dao.content.IContentManager;
+import uk.ac.cam.cl.dtg.segue.search.AbstractFilterInstruction;
+import uk.ac.cam.cl.dtg.segue.search.DateRangeFilterInstruction;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.ac.cam.cl.dtg.isaac.api.Constants.*;
+import static uk.ac.cam.cl.dtg.isaac.api.Constants.DATE_FIELDNAME;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.*;
+
+public class EventNotificationEmailManager {
+    private static final Logger log = LoggerFactory.getLogger(EventNotificationEmailManager.class);
+
+    private final PropertiesLoader properties;
+    private final IContentManager contentManager;
+    private final EventBookingManager bookingManager;
+    private final UserAccountManager userAccountManager;
+    private final EmailManager emailManager;
+    private final PgScheduledEmailManager pgScheduledEmailManager;
+
+
+    /**
+     * This class is required by quartz and must be executable by any instance of the segue api relying only on the
+     * jobdata context provided.
+     */
+    @Inject
+    public EventNotificationEmailManager(final PropertiesLoader properties,
+                                         final IContentManager contentManager,
+                                         final EventBookingManager bookingManager,
+                                         final UserAccountManager userAccountManager,
+                                         final EmailManager emailManager,
+                                         final PgScheduledEmailManager pgScheduledEmailManager) {
+        this.properties = properties;
+        this.contentManager = contentManager;
+        this.bookingManager = bookingManager;
+        this.userAccountManager = userAccountManager;
+        this.emailManager = emailManager;
+        this.pgScheduledEmailManager = pgScheduledEmailManager;
+    }
+
+    public void sendEmailForEvent(IsaacEventPageDTO event, String templateId) throws SegueDatabaseException {
+        List<DetailedEventBookingDTO> eventBookings = bookingManager.adminGetBookingsByEventId(event.getId());
```

Comment:
What booking statuses does this return? I fear this might also email cancelled and other non-confirmed bookings. Should we also be emailing reserved users? With the same email text?

Reply:
Should the method perhaps take a list of booking statuses to use as a filter in the stream below? That might make it both more explicit what the caller wants, and force us to think about it more carefully? Or we could hard-code an explicit list initially so long as we know what we want is the same for both of these emails for now.

Reply:
Thanks for spotting this, it definitely should be filtered - I'm adding a hardcoded 'confirmed' or 'attended' list and double checking the intention with Fergus